### PR TITLE
Shorten boost directory

### PR DESCRIPTION
**Pull request by eloch216**
This PR renames the `boost_1_71_0` directory to `inc`. With this change, BioCro module library R packages can have names up to 16 characters long. I don't think this is very restrictive; most R package names have fewer than 16 characters. Since the `boost` version number is no longer indicated by the directory name, I added a text file that stores it.

If we go through with this change, we might want to also rename the `boost_1_71_0` directory in `biocro-dev` for consistency.

### CODE REVIEW

**Comment by gsrohde** [comment]
The programming part of this looks fine to me.  (I might want to go over the text of some of the templates once you merge into `reduce-potential-merge-conflicts` though.)

I made one additional commit: it strips trailing whitespace from some lines of the license template file and makes a rather minor bug fix to the script (a fix which I added on my branch probably after you grabbed the script from it).

FYI, if you didn't care about allowing the user to be able to use package names longer than 16 characters if they really want to, you could just change the `+` in the regular expression to `{1,15}`, meaning there must be from one to fifteen additional characters in the name after the first one.  (You'd of course want not to hard code the `15`.)

**Review by gsrohde — APPROVED** [review]

**Comment by gsrohde** [comment]
I don't know that it matters much, but I realize now that I probably should have made my commit to the `reduce-potential-merge-conflicts` branch rather than this one.

File: boost_1_71_0/boost_version.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+1.71.0
```
Comment (gsrohde) [inline boost_1_71_0/boost_version.txt]: This version information _is_ in `boost/version.hpp`.  But perhaps you think it less obscure to duplicate it in a plain text file as opposed to a code file.

Comment (eloch216) [reply boost_1_71_0/boost_version.txt]: I didn't know it was in there already. I'll remove this file and merge into the other branch.

**Pull request merged**